components/motor/esp_simplefoc/port/angle_sensor/as5048a.cpp

```diff
@@ -0,0 +1,133 @@
+#include "as5048a.h"
+#include "esp_log.h"
+#include "esp_check.h"
+
+
+#define AS5048A_REG_ANGLE   0x3FFF
+#define PI                  3.14159265358979f
+
+static const char *TAG = "AS5048a";
+
+
+AS5048a::AS5048a(spi_host_device_t spi_host, gpio_num_t sclk_io, gpio_num_t miso_io, gpio_num_t mosi_io, gpio_num_t cs_io)
+{
+    _spi_host = spi_host;
+    _sclk_io = sclk_io;
+    _miso_io = miso_io;
+    _mosi_io = mosi_io;
+    _cs_io = cs_io;
+    is_installed = false;
+}
+
+
+AS5048a::~AS5048a()
+{
+    if (is_installed) {
+        deinit();
+    }
+}
+
+
+void AS5048a::deinit()
+{
+    esp_err_t ret;
+    ret = spi_bus_remove_device(spi_device);
+    ESP_RETURN_ON_FALSE(ret == ESP_OK,, TAG, "SPI remove device fail");
```

Comment:
Here, only the SPI device is closed; it would be better to also release the SPI bus